Reworked paper to address more technical ADASS audience (#4)
lsst-dm/dmtn-209: O3-2.tex

This paper describes our experience with an early release of the RSP to users through our pre-operations program.

\section{The Rubin Interim Data Facility on Google Cloud}

We decided to deploy services involved in our pre-operations program on a commodity cloud services provider as an Interim Data Facility (IDF) to mitigate uncertainty in the final location of our operations-era US Data Facility (USDF).
Following a competitive tender, we selected Google Cloud for this three-year activity to prepare our team for operational readiness and our community for interacting with Rubin data and services.
The IDF is a full stand-alone facility (without dependance on on-premises computing) and is a real high-availability production environment for all services currently oriented to external users. These include the Rubin Science Platform as well as underlying services such as Qserv (our high performance database system) and Butler (data abstraction middleware).

The architecture is depicted in Figure \ref{fig:IDFarch} and leverages the Google Kubernetes Engine (GKE) and Google Cloud Storage (GCS) to scale up and down as needed depending on demand.

> Kubernetes is our reference deployment platform both on-premises and in the Cloud as our services make heavy use of container orchestration patterns and consume containers produced by our build systems based on Jenkins and GitHub Actions,  produces Docker containers for deployment \citep[see e.g.,][]{2018SPIE10707E..09J}.

Reviewer: ```suggestion
Kubernetes is our reference deployment platform both on-premises and in the Cloud as our services make heavy use of container orchestration patterns and consume containers produced by our build systems based on Jenkins and GitHub Actions \citep[see e.g.,][]{2018SPIE10707E..09J}.
```

Besides extensive document at \href{https://dp0-1.lsst.io}{dp0-1.lsst.io} supporting delegates in a number of ways, delegate resources included biweekly virtual ``Delegate Assemblies'' with hands-on demonstrations and breakout rooms to facilitate co-working; a dedicated category in the Rubin Community Forum (\url{https://community.lsst.org}) (based on the Discourse web forum platform) where delegates can ask questions and discuss their DP0 work; and a GitHub repository where delegates can contribute to shared code and notebooks.
GitHub Issues were also used as the primary means for delegates to request technical assistance. All of the delegate pedagogical resources are publicly accessible.

Some aspects of the Community Engagement model as applied to DP0 are designed to scale to thousands of users for the future Data Previews, such as the extensive documentation, the tutorials, and the use of the Community Forum.

> Other aspects, such as the frequent live virtual sessions, would not scale well to thousands of users and were instead specifically designed to ramp up skill levels from novice to intermediate in order to seed expertise with Rubin software in the community.

Reviewer: may not scale well - not sure this is the place for that discussion with CET ..

Author: I didn't write that text.

> Other aspects, such as the frequent live virtual sessions, would not scale well to thousands of users and were instead specifically designed to ramp up skill levels from novice to intermediate in order to seed expertise with Rubin software in the community.

Reviewer: ```suggestion
Other aspects, such as the frequent live virtual sessions, may not scale well to thousands of users and were specifically designed to ramp up skill levels from novice to intermediate in order to seed expertise with Rubin software in the community.
```
